dispatcher_plugin_nb2workflow/queries.py

```diff
@@ -124,20 +125,28 @@ def get_data_server_query(self, instrument, config=None, **kwargs):
     
     def build_product_list(self, instrument, res, out_dir, api=False):
         prod_list = []
+        _output = None
         if out_dir is None:
             out_dir = './'
-        if 'output' in res.json().keys(): # in synchronous mode
-            _o_dict = res.json() 
+        res_content_type = res.headers.get('content-type', None)
+        if res_content_type is not None and res_content_type == 'application/json':
```

Comment:
This works well, of course, but we can think about more explicit way to distinguish the *progress* response vs normal products

Reply:
I agree, it has to be improved. I had thought of adding an additional arg to the `build_product_list` but it would require to update such method for all the other plugins

Reply:
Yes, something less intrusive would be better)

Reply:
by returning a dict containing the response, along with additional entries, in `get_progress_run`, the check can be done in a clean way